### PR TITLE
[CI] Enable custom images to be set in TVM Jenkinsfile

### PR DESCRIPTION
**Pull request by leandron**
Enable custom images to be set in TVM Jenkinsfile:
* This work is needed to enable automatic testing of our newly built Docker images as part of CI
* The default value is set by variables in the same Jenkinsfile and are used when no custom values are provided
* Changes are only expected to be visible in the Jenkins UI

This is also a follow-up on work done in https://github.com/tlc-pack/tlcpack/pull/62, that will allow use to test the images using always the latest from TVM Jenkinsfile.

cc @areusch @jroesch @tqchen @mbrookhart @Mousius for reviews

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -107,6 +122,30 @@ def cancel_previous_build() {
 
 cancel_previous_build()
 
+stage('Prepare') {
+  node('master') {
+    // When something is provided in ci_*_param, use it, otherwise default with ci_*
```
Comment (areusch) [inline Jenkinsfile:127]: just curious why not to use defaultValue instead of doing this? we could rename e.g. ci_cpu to prod_ci_cpu or something.

Comment (leandron) [reply Jenkinsfile:127]: That was my original plan, but Jenkins has these quirks... we can set the default parameter, and that will be parsed/adopted by the UI once. To update that later, it needs to be done via the UI, hence we can't count on these defaults to be set _as code_.

**Comment by leandron** [comment]
CI is failing on random things. Re-triggering CI.

File: Jenkinsfile (hunk 1 above)
Comment (areusch) [reply Jenkinsfile:127]: aha, i see. seems reasonable to me.

**Comment by areusch** [comment]
as a sanity check pushed to staging build: https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/ci-docker-staging/126/pipeline

**Comment by leandron** [comment]
Can you run this again on staging? It seems the job was overtaken by another job. No rush, just a gentle ping to check whether this can go in, to allow the automated Docker image testing to progress.

**Review by Mousius — APPROVED** [review]
Makes sense to me :smile_cat: (pending build ofc)

**Comment by areusch** [comment]
ok just pushing this to staging again just long enough we can see that it's launching containers correctly

**Review by areusch — APPROVED** [review]
awesome! tested by force-push to ci-docker-staging, confirmed:
1. build with no parameters continues to work
2. parameters can be overridden ("Build with Parameters" to manually trigger)
3. parameter override is not somehow sticky (following force-push reverts to no-params build)

i'm good with this one, let's merge as soon as it passes the regular Jenkins flow. don't need to block on ci-docker-staging.

**Comment by leandron** [comment]
OK, so based on @areusch's approval, I'm merging this.

Thanks @Mousius and @areusch!

**Comment by leandron** [comment]
TL;DR:
if you see `No such property: ci_lint_param for class: groovy.lang.Binding`, just force-push your PR again.

----

It looks like for existing PRs, this change on the Jenkins file needs to be reloaded once in CI for each pipeline, and will need at least one re-run to make it work.

In case you see this fail with the error below, **just force-push your PR again, and the error will go away.**

```
groovy.lang.MissingPropertyException: No such property: ci_lint_param for class: groovy.lang.Binding
	at groovy.lang.Binding.getVariable(Binding.java:63)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onGetProperty(SandboxInterceptor.java:270)
	at org.kohsuke.groovy.sandbox.impl.Checker$7.call(Checker.java:353)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:357)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:333)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:333)
	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.getProperty(SandboxInvoker.java:29)
	at com.cloudbees.groovy.cps.impl.PropertyAccessBlock.rawGet(PropertyAccessBlock.java:20)
	at WorkflowScript.run(WorkflowScript:128)
	at ___cps.transform___(Native Method)
	at com.cloudbees.groovy.cps.impl.PropertyishBlock$ContinuationImpl.get(PropertyishBlock.java:74)
...
```
